### DISCUSSION TITLE
How to refresh/refetch a session ?

### DISCUSSION THREAD

**Discussion by zikk** [clerk/javascript#1639] [Support request - DO NOT OPEN HERE]
I am making a mobile app with [capacitor](https://capacitorjs.com/), and when i do am OAuth sigin in and return to the app with deep linking, the session cookies seem to persist because after refreshing the app i get a session

Since the page doesn't initially refresh (which we wouldn't want to happen) we don't get a session. And i don't see a way with the react or vanilla js library to force a network retry to get a session.

**Comment by jescalan** [comment]
https://github.com/orgs/clerk/discussions/3829

**Comment by doug-stewart** [comment]
I had the same question and found this in Clerk's documentation:
https://clerk.com/docs/references/javascript/user#reload

You can call the `reload()` method on the `User` object and it will refetch their data from Clerk, which will have any updates.

So for anyone else who gets directed here from Google, this is likely to be the solution you need.
